### ISSUE TITLE
Possibility to tag/mark/flag visiotors directly from piwik backend

### ISSUE THREAD

**Issue by hpvd** [piwik/piwik#8161]
A possibility to tag/mark/flag visitors manually directly from in the visitorlog with an individual text (simple text  input field)
Should be stored only within piwik as something like a label or addition to visitor ID

And of course it would be helpful to be able to filter via segment-editor for this...

One of several use-cases:
This would be really helpful, to manually mark visits with special characteristics you find during visit analysis when trying to find problems with your website etc.

some of this was already mentioned as a sideline in
https://github.com/piwik/piwik/issues/1839

**Comment by hpvd** [comment]
hmm maybe one do not need a complete new "field" but the possibility to add a custom variable afterwards via the piwiks backend in visitorlog?

**Comment by mattab** [comment]
Thanks for the suggestion @hpvd

**Comment by hpvd** [comment]
@mattlab Would you see this as an piwik function or as a new plugin?

**Comment by mattab** [comment]
@hpvd it could be both, but maybe it would make more sense as a new plugin

**Comment by hpvd** [comment]
to more questions @mattlab:
- do you have any advise/ preference already in mind how this should be implemented from technical pow?
- are there any plans on possibilities how one could vote for / or help with partial sponsoring
to give a new features (at least in a raw but useable version) a good position on the global roadmap?

**Comment by mattab** [comment]
>     do you have any advise/ preference already in mind how this could/should be implemented from technical pow?

Not really, didn't think about this yet. I guess we would need a new dimension for visits (new table column), and then let Piwik user tag the visitors within the Visitor Log. This new dimension could be available as segment.

> are there any plans on possibilities how one could vote for / or help with partial sponsoring to give new features one like a good position on the global roadmap (at least for a raw but usable version)?

unfortunately no, we are very much under resourced and we are focused for the next 6 months on Piwik 3.0: https://github.com/piwik/piwik/milestones/3.0.0

Best way to be heard about features is to post a comment in the issue, because we read all comments :-)

**Comment by hpvd** [comment]
since this feature could be done best directly in visitor log, https://github.com/piwik/piwik/issues/6111should be needed for this too

**Comment by B2Bi** [comment]
Tagging would bring great value to the piwik platform.  The first and most needed tag feature would be to associate a ip address or the "user profile".   Put to a vote from the end users I have to believe this would be one of the top three features they would like to see.  It's my #1.   Keep up the good work
- Tag waiting end user

**Comment by mafrieger** [comment]
+1, tagging would make Piwik even more unique.
We would also appreciate a possibility to vote for features or something like a donation account to give great features a boost....

**Comment by hpvd** [comment]
this would perfectly fit within a standardised approach for actions on tracked data, described in #9182

**Comment by ReessKennedy** [comment]
+1 for this feature!

**Comment by mirabyte** [comment]
+1 as well! Would be really a helpful feature!

**Comment by mattab** [comment]
This feature will maybe need a column in the log_visit table (and other log_* tables) ? Wondering if we can do this now or need to wait for Matomo 4 to introduce the new column to all users. @diosmosis @tsteur

**Comment by tsteur** [comment]
Would need to wait for Matomo 4 probably :( We could maybe use a new log table for it though? be best and what I had roughly in mind for this anyway as you would need to store lots of additional data?

**Comment by mattab** [comment]
by  lots of additional data, are you think we'll need to support multiple columns and not just one?

Btw, maybe we could use Custom Dimensions of "Visit scope" to store the "tagging" information, or is there some disadvantage to using custom dimensions?

**Comment by tsteur** [comment]
> by lots of additional data, are you think we'll need to support multiple columns and not just one?

yes I'm thinking you want to tag people with multiple tags/notes/labels.

**Comment by sgiehl** [comment]
Why do we need to add that to the log_visit table? Can't we use a new table and join that on log_visit. Maybe https://github.com/matomo-org/matomo/pull/14062 might help with that

**Comment by mattab** [comment]
Using a new table to store the multiple tags/notes/labels sounds good to me @sgiehl @tsteur

**Comment by piered** [comment]
I think it would be useful to be able to trace the display of html5 videos inside a page. I tried, but at the moment this activity is not recorded ...
Thank you!

**Comment by mattab** [comment]
Maybe to make this feature happen later we could implement it in a more generic way for example creating a "Custom Dimension of Visitor/User scope" to complement our Visit-scope and Action-scope dimensions? Then when marking users, we'd mark the visitor ID rather than the visit. This more generic feature could also be re-used later by a feedback tool or Net Promoter Score feature.

**Comment by lorimode** [comment]
Ajouté la possibilité de marqué une visite et un visiteur :

Il serait bien de pouvoir marqué un visiteur, ou même un visite, par exemple, pour la retrouvé plus facilement, parce que c'est une visite "modèle" et que je veux pouvoir la revoir et la gardé comme référence, ou bien parce que cette visite c'est mal passée, et que je veux garder le parcours de l'usager pour améliorer mon site...
Bref, plain de possibilité avec cette potentiel fonction, d’où l'idée de l'ajouter !

---trad google---
Ajout de la possibilité de marquer une visite et un visiteur :

Ce serait bien de pouvoir marquer un visiteur, voire une visite par exemple, pour la retrouver plus facilement, car c'est une visite "modèle" et je souhaite pouvoir la revoir et la garder comme référence , ou parce que cette visite s'est mal passée, et que je souhaite conserver le parcours utilisateur pour améliorer mon site...
Bref, beaucoup de possibilités avec cette fonction potentielle, d'où l'idée de l'ajouter !

**Comment by lorimode** [comment]
Et du-coups, cela en est ou ?
---trad google---
And so, where in is that ?
